### ISSUE TITLE
Locations should not be reported in info strings

### ISSUE THREAD

**Issue by jdkato** [vale-cli/vale#248]
Using Vale v2.4.2 here.

This issue needs to have more detail but I may experience something similar with code block and inline code being checked by Vale. Not linked to GitHub action on my side, it also happens locally.

For instance I have ` 41:4     error    Did you really mean 'json'?     Vale.Spelling` when the line 41 is the block start of:
````markdown
```json
"encryptor":{
    "type":"EC",
    "properties":{
        "symmetricCipher":"AES/GCM/NoPadding",
        "ellipticCurve":"secp256r1",
        "nonceLength":"24",
        "sharedKeyLength":"32"
    }
}
```
````

Of course, if I remove the code type on the code block it works for this block, but then fails on the next JSON block. And I require all my blocks to have a code type defined in my Markdown linter to have the correct highlighting, so that's not an option to remove.

Also note that `json` is a word that I have as uppercased `JSON` in an acronym rule that is a copy of the MS one (among other acronyms):

```yaml
extends: conditional
message: "'%s' has no definition."
level: suggestion
ignorecase: true
# Ensures that the existence of 'first' implies the existence of 'second'.
first: '\b([A-Z]{3,5})\b'
second: '(?:\b[A-Z][a-z]+ )+\(([A-Z]{3,5})\)'
# ... with the exception of these:
exceptions:
  ...
  - JSON
  ...
```

But this doesn't change anything.

However if I add `JSON` in the `accepted.txt` I have the following error:
`41:1     error    Use 'JSON' instead of 'json'.   Vale.Terms` meaning `Vale.Terms` looks in the code block.

If I add `json` in my `accepted.txt` Vocab, then I don't have the issues on code blocks but it's not good as I expect writers to use `JSON` everywhere in the text.

This then conflicts with the acronym rule as it says for instance `109:17   error    Use 'json' instead of 'JSON'.   Vale.Terms` and I don't want that, I want the uppercased acronym to be used.

Let me know if you need more testing from me.

_Originally posted by @NicolasMassart in https://github.com/errata-ai/vale/issues/244#issuecomment-706098952_

**Comment by jdkato** [comment]
I moved this to a new issue because it's almost certainly a different issue than #244.

With how Vale handles markup, there's a distinction to be made between the validity of an error and the accuracy of its location -- see https://github.com/errata-ai/vale/issues/95#issuecomment-441308412 for a more detailed explanation of this process. And, in general, issues like these are almost always location-related: it's not that Vale is linting the code block; it's reporting the location incorrectly.

This particular issue is an edge case where the token in question (`json`) can appear verbatim in [info strings](https://github.github.com/gfm/#info-string). The actual error is coming from the following line in [Tessera.md](https://github.com/ConsenSys/doc.tessera/blob/master/docs/HowTo/Configure/Tessera.md#L41):

> Place the wrapped output, `ENC(rJ70hNidkrpkTwHoVn2sGSp3h3uBWxjb)`, in the config json file

It should be fixed in the next release.

**Comment by NicolasMassart** [comment]
Fixing the `json` on line 204  indeed solved this issue on line 41. Thanks.

**Comment by jdkato** [comment]
This should be fixed in v2.4.3.

**Comment by NicolasMassart** [comment]
I confirm 2.4.3 now gives the correct line number, thanks!
